pkg/skr/common/leases/lease.go

```diff
@@ -0,0 +1,110 @@
+package leases
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/cloud-manager/pkg/composed"
+	"github.com/kyma-project/cloud-manager/pkg/skr/runtime/config"
+	v1 "k8s.io/api/coordination/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sync"
+	"time"
+)
+
+type LeaseResult int
+
+const (
+	RenewedLease  LeaseResult = iota
+	AcquiredLease LeaseResult = iota
+	LeasingFailed LeaseResult = iota
+	OtherLeased   LeaseResult = iota
+)
+
+var leaseMutex sync.Mutex
+
+func Acquire(ctx context.Context, cluster composed.StateCluster, resourceName, ownerName types.NamespacedName, prefix string) (LeaseResult, error) {
+	leaseMutex.Lock()
+	defer leaseMutex.Unlock()
+
+	leaseName := getLeaseName(resourceName, prefix)
+	leaseNamespace := resourceName.Namespace
+	holderName := getHolderName(ownerName)
+
+	lease := &v1.Lease{}
+	err := cluster.K8sClient().Get(ctx, types.NamespacedName{Name: leaseName, Namespace: leaseNamespace}, lease)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return LeasingFailed, err
+	}
+	leaseDuration := new(int32)
+
+	*leaseDuration = int32(config.SkrRuntimeConfig.SkrLockingLeaseDuration.Seconds())
+	if lease.Name == "" {
+		lease = &v1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      leaseName,
+				Namespace: leaseNamespace,
+			},
+			Spec: v1.LeaseSpec{
+				HolderIdentity:       &holderName,
+				LeaseDurationSeconds: leaseDuration,
+				AcquireTime:          &metav1.MicroTime{Time: time.Now()},
+				RenewTime:            &metav1.MicroTime{Time: time.Now()},
+			},
+		}
+		err := cluster.K8sClient().Create(ctx, lease)
+		if err != nil {
+			return LeasingFailed, err
+		}
+		return AcquiredLease, nil
+	}
+	if *lease.Spec.HolderIdentity == holderName {
```

Comment:
```suggestion
	if ptr.Deref(lease.Spec.HolderIdentity, "") == holderName {
```
As `HolderIdentity` is optional in `LeaseSpec`, I think we should be safe here and avoid potential nil pointer

Reply:
Done